### PR TITLE
Decompile mail

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/mail.c [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+    u8 messageTop;
+    u8 messageLeft;
+    const struct MailEcWordLayout * linesLayout;
+};
+
+struct MailGfxData
+{
+    const void * pal;
+    const void * tiles;
+    const void * map;
```
Comment (jiangzhengwenjz) [inline src/mail.c]: inconsistent pointer space style

File: src/mail.c [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+    u32 size;
+    u16 textpals[2];
+};
+
+struct MailViewResources {
+    u8 messageLinesBuffer[8][26];
+    u8 authorNameBuffer[12];
+    void (*savedCallback)(void);
+    void (*showMailCallback)(void);
+    struct MailStruct *mail;
```
Comment (jiangzhengwenjz) [inline src/mail.c]: inconsistent pointer space style

File: src/mail.c [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+      	.numRows = 5,
+      	.nameY = 0,
+      	.nameX = 16,
+      	.messageTop = 3,
+      	.messageLeft = 8,
+      	.linesLayout = sLayout_5x2
+    },
+};
+
+void ReadMail(struct MailStruct * mail, void (* savedCallback)(void), bool8 messageExists)
```
Comment (jiangzhengwenjz) [inline src/mail.c]: inconsistent func pointer space style with line 63 & 64. But this style in the header :/

File: src/mail.c [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+    sMailViewResources->copyEasyChatWord = CopyEasyChatWord;
+    sMailViewResources->convertEasyChatWordsToString = ConvertEasyChatWordsToString;
+    if (IS_ITEM_MAIL(mail->itemId))
+    {
+        sMailViewResources->mailType = ITEM_TO_MAIL(mail->itemId);
+    }
+    else
+    {
+        sMailViewResources->mailType = 0;
+        messageExists = 0;
```
Comment (jiangzhengwenjz) [inline src/mail.c]: FALSE

File: src/mail.c (hunk 1 above)
Comment (PikalaxALT) [reply src/mail.c]: Imma keep it. `(void|(struct|union) \w+) \* \w+`, but `v?(u|s|bool)(8|16|32|64) \*\w+`

**Pull request merged**